### PR TITLE
Do not export qesap AWS credential functions

### PR DESCRIPTION
**Pull request by mpagot**
The qesap library has two methods for the creation of the AWS credential files. These two are only used in qesap_prepare_env so they not need to be exported.
Change the unit testing to test them through qesap_prepare_env

- Verification run:

  - HANASR:
      - sle-15-SP4-HanaSr-Azure-Byos-x86_64-Build15-SP4_2023-08-22T04:03:15Z-hanasr_azure_test_sapconf@64bit 
          - http://openqaworker15.qa.suse.cz/tests/216623
          - http://openqaworker15.qa.suse.cz/tests/216627
          - http://openqaworker15.qa.suse.cz/tests/216626
          - http://openqaworker15.qa.suse.cz/tests/216633 :green_circle: conf.yaml generation is fine, no AWS credential part as expected considering it is an Azure test. Deployment failure is in Ansible and is known issue `Timed out waiting for last boot time check `
      - sle-15-SP4-HanaSr-Aws-Byos-x86_64-Build15-SP4_2023-08-22T04:03:15Z-hanasr_aws_test_saptune@64bit
          - http://openqaworker15.qa.suse.cz/tests/216629
          - http://openqaworker15.qa.suse.cz/tests/216630
          - http://openqaworker15.qa.suse.cz/tests/216631 :green_circle:  Deployment ok, code for the AWS credential properly executed http://openqaworker15.qa.suse.cz/tests/216631#step/deploy_qesap_terraform/221

  - QESAP REGRESSION:
      - sle-15-SP4-Qesap-Aws-Payg-x86_64-BuildLATEST_AWS_SLE15_4_PAYG-qesap_aws_saptune_test@64bit
          - http://openqaworker15.qa.suse.cz/tests/216624
          - http://openqaworker15.qa.suse.cz/tests/216636 :green_circle: Deployment ok, code for AWS credential properly executed http://openqaworker15.qa.suse.cz/tests/216636#step/configure/206
      - sle-15-SP5-Qesap-Azure-Payg-x86_64-BuildLATEST_AZURE_SLE15_5_PAYG-qesap_azure_saptune_test@64bit
          - http://openqaworker15.qa.suse.cz/tests/216625
          - http://openqaworker15.qa.suse.cz/tests/216628
          - http://openqaworker15.qa.suse.cz/tests/216632 :green_circle: deployment ok. No AWS credential code executed, as expected as it is an Azure test. Content of the conf.yaml is ok http://openqaworker15.qa.suse.cz/tests/216632/file/configure-qesap_azure.yaml

  - TRENTO:    
    - sle-15-SP4-Trento-PremiumRolling-x86_64-BuildRolling2.1.0-build62.3-trento_test@64bit
          - http://openqaworker15.qa.suse.cz/tests/216634
          - http://openqaworker15.qa.suse.cz/tests/216635 :green_circle: qe-sap-deployment part is ok

### CODE REVIEW

File: lib/qesapdeployment.pm [hunk 1]
```diff
@@ -504,12 +503,19 @@ sub qesap_get_ansible_roles_dir {
...
+
+=back
 =cut
 
 sub qesap_prepare_env {
     my (%args) = @_;
+    croak "Missing mandatory argument 'provider'" unless $args{provider};
     my $variables = $args{openqa_variables} ? $args{openqa_variables} : qesap_get_variables();
-    my $provider = $args{provider};
+    my $provider_folder = lc $args{provider};
```
Comment (mpagot) [inline lib/qesapdeployment.pm:518]: moved here as it was in all the caller

File: t/09_qesapdeployment.t [hunk 2]
```diff
@@ -595,7 +595,7 @@ subtest '[qesap_upload_crm_report]' => sub {
     qesap_upload_crm_report(provider => 'SAND', host => 'SALT');
 
     note("\n  C-->  " . join("\n  C-->  ", @calls));
-    ok 1;
+    ok((any { /.*crm report.*/ } @calls), 'crm report is called');
```
Comment (mpagot) [inline t/09_qesapdeployment.t:598]: a little bit out of topic in this PR but `ok 1;` was really meaningless

**Review by lpalovsky — APPROVED** [review]
LGTM

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

**Review by lilyeyes — APPROVED** [review]
LGTM.

**Pull request merged**